tests/PHPUnit/Integration/DataAccess/LogAggregatorTest.php

```diff
@@ -408,6 +409,25 @@ public function testGenerateQueryWithSegmentShouldUseTmpTableWhenEnabledAndPrima
         $this->assertSame($expected, $query);
     }
 
+    public function testGenerateQuerySwitchesSupportsUncommittedToTrueWhenSupports()
+    {
+        $segment = new Segment('userId==1111', array($this->site->getId()));
+
+        $params = new Parameters($this->site, $this->period, $segment);
+        $this->logAggregator = new LogAggregator($params);
+        $this->logAggregator->allowUsageSegmentCache();
+
+        $this->setSqlRequirePrimaryKeySetting(1);
+
+        $db = Db::get();
+
+        $db->supportsUncommitted = null;
+
+        $this->logAggregator->generateQuery('test, test2', 'log_visit', '1=1', false, '5');
+
```

Comment:
Not unsetting this seems to cause the test errors on mariadb
```suggestion

        $this->setSqlRequirePrimaryKeySetting(0);

```